### DISCUSSION TITLE
页面经过transform: scale后布告牌的点击事件触发不了

### DISCUSSION THREAD

**Discussion by HU-Wenxing-ZuiShuai** [zouyaoji/vue-cesium#382] [Q&A] [answered]
页面经过transform: scale后布告牌的点击事件触发不了

**Comment by zouyaoji** [comment]
布告板的事件是基于cesium的scene.pick来实现的，使用transform之后估计导致cesium的这个api都失效了，具体你可以验证一下，建议不要用transform。

https://github.com/zouyaoji/vue-cesium/blob/dev/packages/composables/use-events/index.ts#L92

**Comment by HU-Wenxing-ZuiShuai** [comment, marked as answer]
下载源码，修改
![image](https://github.com/zouyaoji/vue-cesium/assets/77825182/bbd7ac5e-b75a-4556-a6d2-f56c19e397de)
将cesium重新带包，修改vue-cesium中cesium源,可以解决

**Reply by zouyaoji** [reply]
很强，有类似的需求的人可以参考这个。

**Reply by kankantian** [reply]
这种方案在本地调试可以， 线上发布的时候，依赖包怎么改？

**Reply by kankantian** [reply]
@zouyaoji 有想法将适配方案纳入到下一版本计划中吗

**Reply by HU-Wenxing-ZuiShuai** [reply]
我这的做法是把修改后的源码发布到nginx,然后再main.js种修改cesium源
![image](https://github.com/zouyaoji/vue-cesium/assets/77825182/9d9d2263-a6a1-48cb-9a81-6dde888d0e77)

**Reply by kankantian** [reply]
@HU-Wenxing-ZuiShuai 我打包试了下 发现仍然没有解决问题 可以把你打包的文件发给我测试下吗

**Reply by kankantian** [reply]
下载了cesium源码包, 更改了源码，使用build和release两种方式打包，打开了sourcemap, 在浏览器中调试，发现最后发现引用的还是原本的依赖包， 不知道是哪一步出了问题？不知道是否要删除原有依赖包，然后重新安装？（cesiumPath相应的替换成了）
![image](https://github.com/zouyaoji/vue-cesium/assets/8832463/0d903721-090f-41ad-a928-e066eda01755)

**Comment by HU-Wenxing-ZuiShuai** [comment]
这个是我打包的文件，vue的main,js文件里修改cesium源，不要用vue-cesium自带的cesium包
[CesiumTransform.zip](https://github.com/zouyaoji/vue-cesium/files/13295503/CesiumTransform.zip)

**Reply by kankantian** [reply]
修改了， 估计是打包的问题， 可以分享下打包的过程吗

**Reply by kankantian** [reply]
搞定， 最新版本的cesium源码在编译时默认使用node_models下的@cesium/engine，从而导致修改源码编译未生效。参考了链接：https://blog.csdn.net/u010206379/article/details/130594320
